### PR TITLE
When unless void default

### PR DESCRIPTION
**Pull request by BertvanHoekelen**
- [x] Added or updated tests
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

**Changes**

The default callback for when and unless can can return void.

### CODE REVIEW

File: stubs/EloquentBuilder.stub [hunk 1]
```diff
@@ -437,7 +437,7 @@ class Builder
      *
      * @param  mixed  $value
      * @param  callable($this, mixed): (void|Builder<TModelClass>)  $callback
-     * @param  callable($this, mixed): (null|Builder<TModelClass>)|null  $default
+     * @param  callable($this, mixed): (null|void|Builder<TModelClass>)|null  $default
```
Comment (szepeviktor) [inline stubs/EloquentBuilder.stub:440]: Isn't it void in place of null?
```suggestion
     * @param  callable($this, mixed): (void|Builder<TModelClass>)|null  $default
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ## [Unreleased]
 
 * feat: add support for the `cursorPaginate()` paginator
+* fix: when/unless default values can return void
```
Comment (szepeviktor) [inline CHANGELOG.md:10]:
```suggestion
* fix: when/unless callback for default values can return void
```

File: stubs/EloquentBuilder.stub (hunk 1 above)
Comment (BertvanHoekelen) [reply stubs/EloquentBuilder.stub:440]: Yeah, I was thinking the same. But might be a breaking change if people are already return null now. Let me know If I need to change it.

Comment (szepeviktor) [reply stubs/EloquentBuilder.stub:440]: @canvural I'm out. The framework uses type juggling: `elseif ($default)`.
Please lead the way!

Comment (szepeviktor) [reply stubs/EloquentBuilder.stub:440]: @BertvanHoekelen In PHP reality void is really null 😢 
```php
function noVoidInPhp(): void
{
    echo 'In PHP reality void is really: ';
}

var_dump(noVoidInPhp());
```

Comment (szepeviktor) [reply stubs/EloquentBuilder.stub:440]: Sorry, I'm paralyzed by code-liberalism :)

Comment (canvural) [reply stubs/EloquentBuilder.stub:440]: In this case I think it's better to have something like `(callable($this, mixed): Builder<TModelClass>)|(callable($this, mixed): void)` to say "this callable can return either `void` or `Builder`"

Can you try that? Thanks.

Comment (BertvanHoekelen) [reply stubs/EloquentBuilder.stub]: @canvural Done 💪🏼

**Review by szepeviktor — APPROVED** [review]

**Comment by canvural** [comment]
Thank you. Sorry for the delay!

**Pull request merged**